usr/share/rear/restore/BORG/default/300_load_archives.sh

```diff
@@ -17,17 +17,49 @@ fi
 # Display header.
 LogUserOutput "
 === Borg archives list ===
-Location:   ${BORGBACKUP_HOST:-USB}
-Repository: $BORGBACKUP_REPO
-"
+
+Location:           ${BORGBACKUP_HOST:-USB}
+Repository:         $BORGBACKUP_REPO
+Number of archives: $archive_cache_lines"
 
 # Display BORGBACKUP_ARCHIVE_CACHE file content
 # and prompt user for archive to restore.
 # Always ask which archive to restore (even if there is only one).
 # This gives possibility to abort restore if repository doesn't contain
 # desired archive, hence saves some time.
+
+# Pagination for selecting archives:
+# Show BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER archives at a time, starting
+# with the current ones.
+# If no valid choice is given, cycle through older archives.
+# Enabled by default (BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER=10).
+# To disable pagination set BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER=0.
+
+archive_cache_last_shown=0
+
+# For timestamp output of Borg archives ISO 8601 format is used:
+# YYYY-MM-DDThh:mm:ss, e.g.: 2020-05-26T00:25:00
+
+# When pagination is disabled by the user, show everything
+[[ $BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER -eq 0 ]] \
+    && BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER=$archive_cache_lines
+
 while true ; do
-    LogUserOutput "$( cat -n "$BORGBACKUP_ARCHIVE_CACHE" | awk '{print "["$1"]", $2,"\t"$3,$4,$5}' )"
+    UserOutput ""
+    LogUserOutput "$( cat -n "$BORGBACKUP_ARCHIVE_CACHE" \
+        | tac \
+        | awk '{print "["$1"]", $4 "T" $5, $2 }' \
+        | tail -n +$(( archive_cache_last_shown + 1 )) \
+        | head -n "$BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER" \
+        | tac )"
```

Comment:
If you are not reverting order, I guess you can remove first and last `tac`, right?

Reply:
In that case both `tac` were logically needed. But I reworked the `tail` and `head` calls to be able to remove both `tac` calls and simplify logic.

Commit 6d6059abf9e676c5fba834ad10a7f5a5da26b0e8 tested both with pagination enabled and disabled. Both worked as expected.

Reply:
I'll test it again today,

Thanks for now!